### DISCUSSION TITLE
MariaDB 10.5.12 connection not working

### DISCUSSION THREAD

**Discussion by gerhardmpl** [Budibase/budibase#4581] [Help]
I installed mariadb _10.5.12_ on my budibase host (Debian 11).

```
apt install mariadb-server mariadb-client
mysql_secure_installation
```

and setup a database `budibase`. Not sure about the charset, but that does not make a difference.

```
mariadb -u root -p
CREATE DATABASE budibase DEFAULT CHARACTER SET utf8 COLLATE utf8_unicode_ci;
CREATE USER 'bbdbadmin'@'%' identified by 'PASSWORD';
GRANT ALL ON budibase.* TO 'bbdbadmin'@'%' IDENTIFIED BY 'PASSWORD' WITH GRANT OPTION;
FLUSH PRIVILEGES;
EXIT;
```

Adding the database to budibase works, but neither fetching tables (imported from the Vehicles demo on the CLI) nor creating tables works.
Fetching tables raises a _Error updating datasource schema_ error (maybe related to #4543).
Creating tables hangs in the _Create new table_ dialog (where the `create` button is allways inactive) with no error at all.

Using `172.17.0.1` or `host.docker.internal` as host in the MySQL configuration makes no difference.
Using MariaDB `root` or `bbdbadmin` user does not make a difference as well.

Asking this as a question since it's most likely a misconfiguration and not a bug.

**Comment by gerhardmpl** [comment]
Might indeed be an issue with Maria DB _10.5.12_, since it works with
- mysql  Ver 14.14 Distrib 5.7.37, for Linux (x86_64)
- mysql  Ver 8.0.28 for Linux on x86_64 (MySQL Community Server - GPL)

**Reply by mjashanks** [reply]
Hey @gerhardmpl  - are you able to access logs on the MariaDB end? Would be good to see the error that is happening here.

**Reply by gerhardmpl** [reply]
Sure, just need to reinstall MariaDB when I have access to my playlab this evening.

**Reply by gerhardmpl** [reply]
@mjashanks I set error log verbosity to 4, restarted MariaDB and did a `tail -f /var/log/mysql/error.log` as well as a `tail -f /var/log/mysql/mysql.log`. Fetching tables in `Budibase` returned the known _Error updating datasource schema_ but no entry/changes in either log. Nothing happens in the logs even with log verbosity 9. Any other log I can check?

**Reply by mjashanks** [reply]
ok - thanks for that. I don't think there is anywhere else that you can check. This is one thing that we need to get better at - debugging inside the platform. Thank you - will look into it.
